library/Fission/Internal/MonadDB.hs

```diff
@@ -0,0 +1,38 @@
+module Fission.Internal.MonadDB
+  ( module Fission.Internal.MonadDB.Types
+  , module Fission.Internal.MonadDB.Class
+  , runDBNow
+  , getInner
+  , ensureOne
+  , ensureOneId
+  ) where
+
+import Control.Monad.Logger
+import Control.Monad.Time
+
+import Database.Esqueleto
+
+import RIO.Time
+import RIO hiding (logError)
+
+import Fission.Internal.MonadDB.Types
+import Fission.Internal.MonadDB.Class
+
+runDBNow :: (MonadTime m, MonadDB m) => (UTCTime -> Transaction m a) -> m a
+runDBNow timeTransaction = do
+  now <- currentTime
+  runDB (timeTransaction now)
+
+ensureOneId :: (Exception err, MonadLogger m, MonadThrow m) => err -> [Entity a] -> m (Key a)
+ensureOneId err entities = pure . entityKey =<< ensureOne err entities
```

Comment:
💅 It seems like more often than not we use `return` instead of `pure` maybe useful to just commit to one.

Reply:
actually asked @expede about this same thing a bit ago:
>Yeah, they're the same
I tend to only use return in do blocks, because it's a pun on imperative programming
And pure is just "bring a pure value into this effectful scope"
Which helps me personally signal which notation I mean to be in
But they're totally interchangeable

Reply:
I mean, we _can_ standardize on `pure`. But that needs to be a team decision.

Thoughts?

@dholms @icidasset @bechurch

Reply:
I personally prefer `return`, because it makes more sense to me, eg. you `return` something at the end of a `do` block. Also `pure` feels a bit weird to me because it's the inverse of what functions are usually named. That is, with `pure` the function is named after the function argument. Normally, I think, functions are named after the result of a function, for example `sum` or `add`.

That said, this is totally personal, I wouldn't mind either.

Reply:
To be clear, I have no strong preference here. Whatever as long as we all agree. However, at least some segment of the community has been slowly moving away from `return`. IMO it makes sense in `do` notation (i.e. where I tend to use it). The argument against is that it means something very different from `return` in any other language. For instance:

```haskell
do
  foo <- runSomething
  bar <- return otherThing
  baz <- return evenMore
  return (foo + bar + baz)
```

You'd normally expect `return` to exit out of a function early.

We can't totally get rid of `pure`, since it's a required method for `Applicative`.

But yeah, sure, we can do `return` everywhere 👍